cdap-ui/app/directives/dag-plus/my-dag-ctrl.js

```diff
@@ -1015,30 +1051,11 @@ angular.module(PKG.name + '.commons')
     vm.cleanUpGraph = function () {
       if ($scope.nodes.length === 0) { return; }
 
-      // If graph has a condition node, then have to make sure/rearrange connections from that
-      // node so that the 'true' connection always comes before the 'false' one, otherweise
-      // the algorithm will mess up
-      if (conditionNodes.length > 0) {
-        let swappedConnections = false;
+      let { connections, swappedConnections } = HydratorPlusPlusCanvasFactory.orderConditionAlertErrorConnections($scope.connections, conditionNodes, alertNodes, errorNodes);
```

Comment:
I am not sure about the name of this function. If we introduce a new node tomorrow this will force us to either rename this function or write a new one. Since `orderConditionAlertErrorConnections` is specifically checking for ordering of conditions, alerts and error nodes and are in separate functions, why not modify the `orderConnections` function to accept nodes and connections and pass on the nodes to these individual function to order specific node types? Provided if we can identify a node to be a connection node or alert node from the `$scope.nodes` array.